TILE_AUTHOR_GUIDE.md

```diff
@@ -0,0 +1,598 @@
+
+#### Default credentials file
+
+You can add a default credentials file to `~/.kiln/credentials.yml` so you don't need to pass variables flags everywhere.
+Don't do this with production creds but if you have credentials you can safely write to your disk, consider using this functionality.
+The file can look like this
+```yaml
+# GitHub BOSH release tarball release sources credentials
+github_token: some-token
+
+# S3 release BOSH release tarball source credentials
+aws_secret_access_key: some-key
+aws_access_key_id: some-id
+
+# Artifactory BOSH release tarball release source credentials
+artifactory_username: some-username
+artifactory_password: some-password
+```
+
+### Release Compilation
+
+_WORK IN PROGRESS EXAMPLE_
+
+```sh
+# create a tile with the releases you want compiled
+kiln bake
+
+# Add an S3 (or Artifactory) release source to your Kilnfile
+yq -i '.release_sources = [{"type": "s3", "id": "my_compiled_release_bucket", bucket": "some_bucket", "publishable": true, "access_key_id": "some_id", "secret_access_key": "some_id"}] + .release_sources' Kilnfile
+
+# claim a bosh director and configure the environment
+smith claim -p us_4_0
+eval "$(smith om)"
+eval "$(smith bosh)"
+
+# deploy your Product (the commands should look something like this)
+om upload-product --product=tile.pivotal
+om configure-product --config=simple_config.yml
+om apply-changes --product-name=my-tile-name
+
+# Download Compiled BOSH Releases from the BOSH Director and Upload them to the S3 Bucket or Artifactory
+kiln cache-compiled-releases --upload-target-id=my_compiled_release_bucket --name=hello
+
+# Commit and push the changes to Kilnfile.lock
+git add -p Kilnfile.lock
+git commit -m "compile BOSH Releases with $(yq '.stemcell_criteria.os' Kilnfile.lock)/$(yq '.stemcell_criteria.version' Kilnfile.lock)"
+git push origin HEAD
+```
+
+### Temporary BOSH Release Tarball Locking
+
+This functionality is likely not helpful for tile authors who only package their own BOSH releases or for those who only package a few BOSH releases.
+
+If you need to pause BOSH Release bumps in your Kilnfile.lock,
+you can execute `kiln glaze`.
+It sets the BOSH release version (constraint) fields to the semver from the Kilnfile.lock.
+[This command has a PR to make it more helpful for TAS/IST/TASW. See the PR here](https://github.com/pivotal-cf/kiln/pull/406).
+This effectively pins releaess to block Dependabot updates.
+PPE uses this command for TAS/IST/TASW prior to new major versions.
+
+## Stemcell Version Management
+
+`kiln find-stemcell-version` and `kiln update-stemcell`
+
+Find the latest stemcell releases on PivNet. They behave similarly to the bosh release commands above.
+
+_If I remember right, the find-stemcell-version command has a bug where the stemcell criteria version in the Kilnfile is not respected and the result of the command is always the latest version._
+
+## Tile Release Note Generation
+
+If you GitHub BOSH Release tarball sources,
```

Comment:
I found this sentence a bit hard to read, for me it seems to be missing a verb.

Reply:
Thanks for the catch. I also broke the sentence up a bit.